### PR TITLE
Defines other activation functions

### PR DESCRIPTION
**Pull request by marceloreichert**

### CODE REVIEW

File: lib/neural_network/activation.ex [hunk 1]
```diff
@@ -0,0 +1,27 @@
+defmodule NeuralNetwork.Activation do
```
Comment (kblake) [inline lib/neural_network/activation.ex]: This is an awesome addition!

File: test/neural_network/activation_test.exs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+defmodule NeuralNetwork.ActivationTest do
```
Comment (kblake) [inline test/neural_network/activation_test.exs]: Thanks for adding tests! ❤️

**Comment by kblake** [comment]
@marceloreichert 

I super appreciate you looking at the code and adding the activation type functionality. 😄 😄 

Tiny thought... what is your opinion on the activation type atoms (`:softmax`, `:sigmoid`, `:relu`) defined as module attributes in `Activation` with associated functions to retrieve them? Then used on lines such as this: `Network.activate(Activation.sigmoid(), sample.input)`. Not a big deal either way, just curious what you think.

**Comment by marceloreichert** [comment]
@kblake ,
I think I understand you. 

But my PR is incomplete, because I believe that we need to define the activation function (relu, tanh, ...) to hidden layers when creating the Network. 

`Network.start_link([1,2], 'relu')`
Something like this. What do you think?

**Comment by kblake** [comment]
Yes, that makes more sense to pass in via the `start_link`. 👍 
That way we can easily support taking in the activation type via mix task arguments. Love it!

**Comment by marceloreichert** [comment]
@kblake Now I believe it is complete. What do you think?

**Comment by kblake** [comment]
Ya, the activation functionality is a great addition. Thank you. Merging...

**Pull request merged**